Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2678 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+import boto3
+import json
+import re
+from datetime import datetime, date
+from botocore.config import Config
+import urllib3.util
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+"""PARAMETERS"""
+AWS_DEFAULT_REGION = demisto.params().get('defaultRegion')
+AWS_roleArn = demisto.params().get('roleArn')
+AWS_roleSessionName = demisto.params().get('roleSessionName')
+AWS_roleSessionDuration = demisto.params().get('sessionDuration')
+AWS_rolePolicy = None
+AWS_access_key_id = demisto.params().get('access_key')
+AWS_secret_access_key = demisto.params().get('secret_key')
+VERIFY_CERTIFICATE = not demisto.params().get('insecure', True)
+if not demisto.params().get('proxy', False):
+    config = Config(
+        retries=dict(
+            max_attempts=10
+        )
+    )
+else:
+    config = None
+
+
+"""HELPER FUNCTIONS"""
+
+
+def aws_session(service='ec2', region=None, roleArn=None, roleSessionName=None, roleSessionDuration=None,
+                rolePolicy=None):
+    kwargs = {}
+    if roleArn and roleSessionName is not None:
+        kwargs.update({
+            'RoleArn': roleArn,
+            'RoleSessionName': roleSessionName,
+        })
+    elif AWS_roleArn and AWS_roleSessionName is not None:
+        kwargs.update({
+            'RoleArn': AWS_roleArn,
+            'RoleSessionName': AWS_roleSessionName,
+        })
+
+    if roleSessionDuration is not None:
+        kwargs.update({'DurationSeconds': int(roleSessionDuration)})
+    elif AWS_roleSessionDuration is not None:
+        kwargs.update({'DurationSeconds': int(AWS_roleSessionDuration)})
+
+    if rolePolicy is not None:
+        kwargs.update({'Policy': rolePolicy})
+    elif AWS_rolePolicy is not None:
+        kwargs.update({'Policy': AWS_rolePolicy})
+    if kwargs and AWS_access_key_id is None:
+
+        if AWS_access_key_id is None:
+            sts_client = boto3.client('sts')
+            sts_response = sts_client.assume_role(**kwargs)
+            if region is not None:
+                client = boto3.client(
+                    service_name=service,
+                    region_name=region,
+                    aws_access_key_id=sts_response['Credentials']['AccessKeyId'],
```

Comment:
are we so sure all this values will be in the response? maybe we need to check first and return a meaningful error?

Reply:
AWS standardises all returned responses and does not break backwards compatibility.